src/vm/gdbjit.cpp

```diff
@@ -1886,13 +1884,15 @@ static void fixLineMapping(SymbolsInfo* lines, unsigned nlines)
     int prevLine = 0;
     for (int i = 0; i < nlines; ++i)
     {
+        if (lines[i].lineNumber == HiddenLine)
+            continue;
         if (lines[i].ilOffset == ICorDebugInfo::PROLOG) // will be fixed in next step
         {
             prevLine = 0;
         }
         else
         {
-            if (lines[i].lineNumber == 0 || lines[i].lineNumber == HiddenLine)
+            if (lines[i].lineNumber == 0)
             {
                 if (prevLine != 0)
```

Comment:
This `if` is not necessary. If you leave it out, if the prevLine was 0, it would assign zero to the lines[i].lineNumber and since it was already zero, it would not change.